tests/api/api.spec.ts

```diff
@@ -745,5 +744,60 @@ describe("@dataform/api", () => {
         })
       );
     });
+
+    it("execute_with_cancel", async () => {
+      const TEST_GRAPH: protos.IExecutionGraph = protos.ExecutionGraph.create({
+        projectConfig: {
+          warehouse: "bigquery",
+          defaultSchema: "foo",
+          assertionSchema: "bar"
+        },
+        warehouseState: {
+          tables: []
+        },
+        nodes: [
+          {
+            name: "node1",
+            dependencies: [],
+            tasks: [
+              {
+                type: "statement",
+                statement: "some statement"
+              }
+            ],
+            type: "table",
+            target: {
+              schema: "schema1",
+              name: "target1"
+            },
+            tableType: "table"
+          }
+        ]
+      });
+      const mockedDbAdapter = mock(BigQueryDbAdapter);
+      when(mockedDbAdapter.prepareSchema(anyString())).thenResolve(null);
+      let wasCancelled = false;
+      // This will deliberately never return.
+      when(mockedDbAdapter.execute(TEST_GRAPH.nodes[0].tasks[0].statement)).thenReturn(
+        new Bluebird((resolve, reject, onCancel) => {
+          onCancel(() => {
+            wasCancelled = true;
+          });
+        })
+      );
+
+      const runner = new Runner(instance(mockedDbAdapter), TEST_GRAPH);
+      const promise = runner.execute();
+      // Push this to the back of the "queue" so we can await the promise first.
+      setTimeout(() => runner.cancel(), 10);
```

Comment:
i don't think i understand what this is doing, can you clarify? it seems like the intention is to allow execution to be running (in the background, triggered by runner.execute()), give it "some time" (10ms), and then cancel? what's the reasoning for waiting 10ms?

Reply:
I've added a better comment here hopefully.
If we don't wait the promise never actually starts, the onCancel handler above never get's set up so we don't test what we actually want to.